### PR TITLE
Remove arrow uasge in FieldData

### PR DESCRIPTION
**Pull request by xige-16**
issue: #20995 
/kind improvement
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/core/src/storage/FieldData.h [hunk 1]
```diff
@@ -18,56 +18,164 @@
...
+    virtual int64_t
+    GetDataSize() const = 0;
 
-    ~FieldData() = default;
+    virtual int64_t
+    GetDim() const = 0;
 
     DataType
-    get_data_type() const {
+    GetDataType() const {
```
Comment (czs007) [inline internal/core/src/storage/FieldData.h:60]: The function naming style should be unified.
better to keep get_data_type()

File: internal/core/src/storage/FieldData.h [hunk 2]
```diff
@@ -18,56 +18,164 @@
...
+
+    virtual void
+    FillFieldData(const void* source, ssize_t element_count) = 0;
+
+    virtual void
+    FillFieldData(const std::shared_ptr<arrow::Array> array) = 0;
 
-    explicit FieldData(std::shared_ptr<arrow::Array> raw_data, DataType data_type);
+    virtual const void*
+    Data() const = 0;
```
Comment (czs007) [inline internal/core/src/storage/FieldData.h:48]: The function naming style should be unified

File: internal/core/src/storage/FieldData.h [hunk 3]
```diff
@@ -18,56 +18,164 @@
...
-class FieldData {
+class FieldDataBase {
  public:
-    explicit FieldData(const Payload& payload);
+    explicit FieldDataBase(DataType data_type) : data_type_(data_type) {
+    }
+    virtual ~FieldDataBase() = default;
+
+    virtual void
+    FillFieldData(const void* source, ssize_t element_count) = 0;
```
Comment (czs007) [inline internal/core/src/storage/FieldData.h:42]: The function naming style should be unified. please use snake style

File: internal/core/src/storage/MinioChunkManager.cpp [hunk 4]
```diff
@@ -327,25 +327,22 @@ MinioChunkManager::GetObjectBuffer(const std::string& bucket_name,
...
-    } else {
-        memcpy(buf, ss.str().data(), size);
-    }
-    return realSize;
+    return size;
 }
 
 std::vector<std::string>
 MinioChunkManager::ListObjects(const char* bucket_name, const char* prefix) {
+    std::cout << "list objects, bucket name = " << bucket_name << ", prefix = " << prefix << std::endl;
```
Comment (czs007) [inline internal/core/src/storage/MinioChunkManager.cpp:345]: Is this on purpose?

**Comment by xiaofan-luan** [comment]
Most part lgtm.
Is there a seperate pr for payload reader?

File: internal/core/src/storage/Util.cpp [hunk 5]
```diff
@@ -26,11 +26,11 @@
...
-ReadMediumType(PayloadInputStream* input_stream) {
-    AssertInfo(input_stream->Tell().Equals(arrow::Result<int64_t>(0)), "medium type must be parsed from stream header");
+ReadMediumType(BinlogReaderPtr reader) {
+    AssertInfo(reader->Tell() == 0, "medium type must be parsed from stream header");
     int32_t magic_num;
-    auto ret = input_stream->Read(sizeof(magic_num), &magic_num);
-    AssertInfo(ret.ok(), "read input stream failed");
+    auto ret = reader->Read(sizeof(magic_num), &magic_num);
+    AssertInfo(ret.ok(), "read binlog failed");
     if (magic_num == MAGIC_NUM) {
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/core/src/storage/Util.cpp:34]: The local disk file format needs to be redesigned, and this method has not been used to read local file for now

[the remark this answers is not in the document]
Comment (xige-16) [reply internal/core/src/storage/Util.cpp:34]: > How do we ensure local disk first int is not MAGIC_NUM?

The local disk file format needs to be redesigned, and this method has not been used to read local file for now

**Comment by xige-16** [comment]
> Most part lgtm. Is there a seperate pr for payload reader?

Yes

**Pull request merged**